app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -34,48 +81,156 @@ export function KubeconfigButton({ environments }: Props) {
+      setShowDialog(false);
+    } catch (e) {
+      notifications.error('Failed downloading kubeconfig file', e as Error);
+    }
   }
-  return environments.some((env) => isKubernetesEnvironment(env.Type));
 }
 
-async function showKubeconfigModal(environments: Environment[]) {
-  const kubeEnvironments = environments.filter((env) =>
+export interface KubeconfigPormptProps {
+  ksExpiry: string;
+  selection: Set<number>;
+  envQueryParams: EnvironmentsQueryParams;
+}
+const storageKey = 'home_endpoints';
+
+export function KubeconfigPormpt({
+  ksExpiry,
+  selection,
+  envQueryParams,
+}: KubeconfigPormptProps) {
+  const [page, setPage] = useState(1);
+  const [pageLimit, setPageLimit] = usePaginationLimitState(storageKey);
+  const [checked, setChecked] = useState(false);
+  const [checkAll, setCheckAll] = useState(false);
+  const { environments, totalCount } = useEnvironmentList(
+    { page, pageLimit, ...envQueryParams },
+    true
+  );
+  const kubeEnvs = environments.filter((env) =>
     isKubernetesEnvironment(env.Type)
   );
-  const options = kubeEnvironments.map((environment) => ({
-    text: `${environment.Name} (${environment.URL})`,
-    value: `${environment.Id}`,
-  }));
-
-  let expiryMessage = '';
-  try {
-    expiryMessage = await kcService.expiryMessage();
-  } catch (e) {
-    notifications.error('Failed fetching kubeconfig expiry time', e as Error);
+  const selected = Array<boolean>();
+  for (let i = 0, len = kubeEnvs.length; i < len; i += 1) {
+    selected.push(selection.has(kubeEnvs[i].Id));
   }
 
-  confirmKubeconfigSelection(
-    options,
-    expiryMessage,
-    async (selectedEnvironmentIDs: string[]) => {
-      if (selectedEnvironmentIDs.length === 0) {
-        notifications.warning('No environment was selected', '');
-        return;
+  return (
+    <div className="modal-body">
+      <form className="bootbox-form">
+        <div className="bootbox-prompt-message">
+          <p>
+            Select the kubernetes environment(s) to add to the kubeconfig file.{' '}
+            <br />
+            {ksExpiry}
+          </p>
+        </div>
+        <div className="form-check">
+          <label>
+            <input
+              type="checkbox"
+              checked={isCheckAll()}
+              onChange={() => onCheckAll()}
+            />{' '}
+            Select all (in this page)
+          </label>
+        </div>
+      </form>
+      <div className="datatable">
+        <div className="bootbox-checkbox-list">
+          {kubeEnvs.map((env, index) => (
+            <div className="form-check" key={env.Id}>
+              <label>
+                <input
+                  type="checkbox"
+                  checked={selected[index]}
+                  onChange={() => onCheck(env.Id)}
+                />{' '}
+                {env.Name} ({env.URL})
+              </label>
+            </div>
+          ))}
+        </div>
+        <div className="footer" style={{ backgroundColor: 'white' }}>
```

Comment:
use css modules, not inline css

Reply:
I have defined a new css module in KubeconfigButton.css

Reply:
module is used `KubeconfigButton.module.css`, and imported using `import styles from 'KubeconfigButton.module.css'`

then you have access to the class `styles.footer`

Reply:
I think this will make it complex. `css module` will rename the footer and sub controls' css as below will not work:
.datatable .footer .infoBar
.datatable .footer .paginationControls
.datatable .footer .paginationControls .limitSelector 
.datatable .footer .paginationControls .pagination

Reply:
Finally, I use `footer` directly. Not change any styles